### ISSUE TITLE
Allow custom size accessor in scatter plots?

### ISSUE THREAD

**Issue by Frozenlock** [dc-js/dc.js#1346]
Scatter plots with more than 1000 items are slow.
![image](https://user-images.githubusercontent.com/1152041/31154063-8ad1435c-a872-11e7-9347-e210847aeff9.png)

One solution to this is to group the items, but this kills the visual feedback given by a scatter plot.

A possible way to get around this would be to apply some bubble-like logic by allowing the size of a symbol to reflect the number of items it represents.

Unfortunately, it appears the [size accessor is overwritten by dc.js](https://github.com/dc-js/dc.js/blob/master/web/docs/api-latest.md#scatterplotcustomsymbolcustomsymbol--string--function--scatterplot) and the provided `symbolSize` method doesn't accept a function.

**Comment by cmorgan** [comment]
I saw an interesting article on using three.js when > 10k data points.
http://blog.fastforwardlabs.com/2017/10/04/using-three-js-for-2d-data-visualization.html
But this seems like overkill for 1k scatter

**Comment by gordonwoodhull** [comment]
Good point @cmorgan, some kind of canvas-based scatterplot would fix this easily (#811). One possibility is to connect Chart.js into the [dc.js chart registry](http://dc-js.github.io/dc.js/docs/html/dc.chartRegistry.html).

I looked into @Frozenlock's proposed solution a little bit but couldn't figure it out either. The scatter plot code is a bit convoluted due to the fact that it's taking [x,y] keys and then cramming them into the coordinate grid model, where keys are X and values are Y. If there is a simple way to allow functions for symbolSize, I'm certainly open to the idea.
